### DISCUSSION TITLE
[Support]: Camera Storage out of sync?

### DISCUSSION THREAD

**Discussion by johnrobertcobbold** [blakeblackshear/frigate#13338] [General Support] [answered]
### Describe the problem you are having

I recently noticed that my Frigate installation that has been working very well was struggling. A look through the logs showed me:

```
OSError: [Errno 28] No space left on device: '/media/frigate/clips/front_yard_camera-1724606206.952738-fdx891.jpg'
```

```
warning
2024-08-25 18:18:03
frigate.record.maintainer
Unable to keep up with recording segments in cache for front_yard_camera. Keeping the 6 most recent segments out of 7 and discarding the rest...
```

OK, so not enough space on the disk? I looked at the storage page:

![image](https://github.com/user-attachments/assets/6d7a3440-3210-41e7-9734-787127651878)

Full camera storage (32 Go) but loads of space on Recordings ?

I used to store all Frigate's videos on the Raspberry Pi 32 Go SD card before recently mounting  an external 500 Go HD

![image](https://github.com/user-attachments/assets/fa250ff6-95ac-42b4-b005-483950eb0ea6)

Would I be right in saying that the camera's storage is still set on 32 Go and had not updated so this is somehow conflicting with the total Recordings space which did update?

Here in case is a `df -h`:

![image](https://github.com/user-attachments/assets/30329524-ea58-4ddb-b617-8dac43640619)

### Version

0.14.0-da913d8

### Frigate config file

```yaml
-
```

### Relevant Frigate log output

```shell
-
```

### Relevant go2rtc log output

```shell
-
```

### FFprobe output from your camera

```shell
-
```

### Frigate stats

```json
-
```

### Operating system

HassOS

### Install method

HassOS Addon

### Object Detector

Coral

### Network connection

Wireless

### Camera make and model

TP-Link Tapo C200

**Comment by NickM-27** [comment]
the space shown is from the database, so you either have recordings that are on the file system but not logged or you have something else. One user found that the recordings were included in their backup and that is what was causing the problem

**Comment by johnrobertcobbold** [comment]
Hi Nick,

Media used to previously be stored in `/opt/frigate/media` so I just had a look:

```
pwd
/opt/frigate/media
raspberrypi:/opt/frigate/media $ du -sh
125M 
```

(I guess I  can just delete that folder by the way?)

Could it be that the database thinks that the camera storage is still limited to 32Go whereas in the meantime, that external hard disk was mounted and used in the volumes as `- /media/passport/frigate/media/:/media/frigate` ?

**Reply by NickM-27** [reply]
No, the database has nothing to do with the total amount of storage available on the disk.

**Reply by johnrobertcobbold** [reply]
All right, so it's not the database.

Current space taken by the volume (docker-compose volumes: /media/passport/frigate/media/:/media/frigate)

```
/media/passport $ du -sh
37G  
```

Don't you think it's an interesting coincidence that the camera storage would (still) be capped at 31.61 GiB / 31.61 GiB when that is the size of the SD card on which the Raspberry runs and on which Frigate previously used (`/opt/frigate/media`) for media before migrating to the external HD ?

**Reply by NickM-27** [reply]
when you edited your docker file, what did you run to apply it and restart frigate?

**Reply by johnrobertcobbold** [reply]
According to my notes, it was back on May 28th that I did the following:

```
sudo mkdir /media/passport
sudo mount /dev/sda1 /media/passport
```
I did not note how I restarted Frigate but I imagine that I would have updated the volumes in my docker-compose to ` - /media/passport/frigate/media/:/media/frigate` before doing a`docker-compose up -d`.

**Reply by NickM-27** [reply]
right, as long as docker-compose up -d was run then it should have been applied. We'd need to see a full copy of the logs then

**Reply by johnrobertcobbold** [reply]
Here we go, I hope that this helps !

(I used the API to retrieve the logs: `http://192.168.35.5:5000/api/logs/frigate?start=-500`)


```
    "2024-08-25 18:16:25.846949379  [INFO] Preparing Frigate...",
    "2024-08-25 18:16:25.983527722  [INFO] Starting Frigate...",
    "2024-08-25 18:16:27.560556582  [2024-08-25 18:16:27] frigate.app                    INFO    : Starting Frigate (0.14.0-da913d8)",
    "2024-08-25 18:16:27.560829730  [2024-08-25 18:16:27] frigate.util.config            INFO    : Checking if frigate config needs migration...",
    "2024-08-25 18:16:27.586804012  [2024-08-25 18:16:27] frigate.util.config            INFO    : frigate config does not need migration...",
    "2024-08-25 18:16:27.984858809  [2024-08-25 18:16:27] frigate.util.services          WARNING : Did not detect hwaccel, using a GPU for accelerated video decoding is highly recommended",
    "2024-08-25 18:16:28.018761543  [2024-08-25 18:16:28] peewee_migrate.logs            INFO    : Starting migrations",
    "2024-08-25 18:16:28.020028007  [2024-08-25 18:16:28] peewee_migrate.logs            INFO    : There is nothing to migrate",
    "2024-08-25 18:16:28.239882332  [2024-08-25 18:16:28] frigate.app                    INFO    : Recording process started: 366",
    "2024-08-25 18:16:28.253124584  [2024-08-25 18:16:28] frigate.app                    INFO    : Recording process started: 368",
    "2024-08-25 18:16:28.257248736  [2024-08-25 18:16:28] frigate.app                    INFO    : go2rtc process pid: 110",
    "2024-08-25 18:16:28.288234467  [2024-08-25 18:16:28] detector.coral                 INFO    : Starting detection process: 395",
    "2024-08-25 18:16:28.293730361  [2024-08-25 18:16:28] frigate.detectors.plugins.edgetpu_tfl INFO    : Attempting to load TPU as usb",
    "2024-08-25 18:16:30.985208485  [2024-08-25 18:16:28] frigate.app                    INFO    : Output process started: 397",
    "2024-08-25 18:16:30.986720430  [2024-08-25 18:16:28] frigate.app                    INFO    : Camera processor started for front_yard_camera: 426",
    "2024-08-25 18:16:30.986892264  [2024-08-25 18:16:28] frigate.app                    INFO    : Capture process started for front_yard_camera: 428",
    "2024-08-25 18:16:30.986920375  [2024-08-25 18:16:28] frigate.app                    INFO    : Audio process started: 435",
    "2024-08-25 18:16:30.990655878  [2024-08-25 18:16:30] frigate.detectors.plugins.edgetpu_tfl INFO    : TPU found",
    "2024-08-25 18:16:51.976057516  Exception in thread detected_frames_processor:",
    "2024-08-25 18:16:51.976076849  Traceback (most recent call last):",
    "2024-08-25 18:16:51.976078312    File \"/usr/lib/python3.9/threading.py\", line 954, in _bootstrap_inner",
    "2024-08-25 18:16:51.976339535      self.run()",
    "2024-08-25 18:16:51.976355016    File \"/opt/frigate/frigate/object_processing.py\", line 1179, in run",
    "2024-08-25 18:16:51.976558220      camera_state.update(",
    "2024-08-25 18:16:51.976563702    File \"/opt/frigate/frigate/object_processing.py\", line 724, in update",
    "2024-08-25 18:16:51.976682794      c(self.name, removed_obj, frame_time)",
    "2024-08-25 18:16:51.976687924    File \"/opt/frigate/frigate/object_processing.py\", line 940, in end",
    "2024-08-25 18:16:51.976828776      with open(",
    "2024-08-25 18:16:51.976842072  OSError: [Errno 28] No space left on device: '/media/frigate/clips/front_yard_camera-1724606206.952738-fdx891.jpg'",
    "2024-08-25 18:18:03.342617342  [2024-08-25 18:18:03] frigate.record.maintainer      WARNING : Unable to keep up with recording segments in cache for front_yard_camera. Keeping the 6 most recent segments out of 7 and discarding the rest...",
    "2024-08-25 18:18:13.349266923  [2024-08-25 18:18:13] frigate.record.maintainer      WARNING : Unable to keep up with recording segments in cache for front_yard_camera. Keeping the 6 most recent segments out of 7 and discarding the rest...",
    "2024-08-25 18:18:23.356723733  [2024-08-25 18:18:23] frigate.record.maintainer      WARNING : Unable to keep up with recording segments in cache for front_yard_camera. Keeping the 6 most recent segments out of 7 and discarding the rest...",
    "2024-08-25 18:18:33.364482329  [2024-08-25 18:18:33] frigate.record.maintainer      WARNING : Unable to keep up with recording segments in cache for front_yard_camera. Keeping the 6 most recent segments out of 7 and discarding the rest...",
    "2024-08-25 18:18:43.366642148  [2024-08-25 18:18:43] frigate.record.maintainer      WARNING : Unable to keep up with recording segments in cache for front_yard_camera. Keeping the 6 most recent segments out of 7 and discarding the rest...",
    "2024-08-25 18:18:53.375433092  [2024-08-25 18:18:53] frigate.record.maintainer      WARNING : Unable to keep up with recording segments in cache for front_yard_camera. Keeping the 6 most recent segments out of 7 and discarding the rest...",
    "2024-08-25 18:19:03.382123005  [2024-08-25 18:19:03] frigate.record.maintainer      WARNING : Unable to keep up with recording segments in cache for front_yard_camera. Keeping the 6 most recent segments out of 7 and discarding the rest...",
    "2024-08-25 18:19:13.391720501  [2024-08-25 18:19:13] frigate.record.maintainer      WARNING : Unable to keep up with recording segments in cache for front_yard_camera. Keeping the 6 most recent segments out of 7 and discarding the rest...",
    "2024-08-25 18:19:23.400287153  [2024-08-25 18:19:23] frigate.record.maintainer      WARNING : Unable to keep up with recording segments in cache for front_yard_camera. Keeping the 6 most recent segments out of 7 and discarding the rest...",
    "2024-08-25 18:19:33.407263516  [2024-08-25 18:19:33] frigate.record.maintainer      WARNING : Unable to keep up with recording segments in cache for front_yard_camera. Keeping the 6 most recent segments out of 7 and discarding the rest...",
    "2024-08-25 18:19:43.415855592  [2024-08-25 18:19:43] frigate.record.maintainer      WARNING : Unable to keep up with recording segments in cache for front_yard_camera. Keeping the 6 most recent segments out of 7 and discarding the rest...",
    "2024-08-25 
```
then it just keeps on and on with those warnings...

**Reply by NickM-27** [reply]
okay, you should get a shell in the container and run `df -h`

**Reply by johnrobertcobbold** [reply]
```
docker exec -it frigate sh
# df -h
Filesystem      Size  Used Avail Use% Mounted on
overlay          29G   18G   11G  63% /
tmpfs            64M     0   64M   0% /dev
shm              64M  4.0M   60M   7% /dev/shm
/dev/mmcblk0p2   29G   18G   11G  63% /config
tmpfs           954M  7.3M  947M   1% /tmp/cache
/dev/sda1       466G   37G  429G   8% /media/frigate
```

It seems to be ok ?

**Reply by NickM-27** [reply]
probably worth doing a full host restart

**Reply by johnrobertcobbold** [reply]
I did a `sudo reboot` to reboot the Raspberry Pi 5, restarted all containers with Portainer... nothing, still the same 🤷🏻‍♂️
![image](https://github.com/user-attachments/assets/05684f5d-9d4f-4941-9fe8-dc4e7b51bd9a)

**Reply by NickM-27** [reply]
Frigate sees the size is different, the logs say that it is the kernel itself which says the filesystem is full.

**Reply by johnrobertcobbold** [reply]
Yes, it was unrelated to the size as I tried changing my Frigate settings to retain less videos which indeed reduced the camera's occupied storage but still lead to the same problem. I wanted to then check the disk Inodes but could not due to the HDD's formatting (fat32). I have just reformatted the disk using my Raspberry Pi to ext4 and will see if this changes anything. This seems unrelated to Frigate although I did come across other Frigate users facing a similar problem. I will report here in case this can help other users (if I can work it out that is). Thanks for your help and time Nick.

**Reply by johnrobertcobbold** [reply, marked as answer]
Some feedback as promised, since I reformatted my HDD as ext4, I no longer have the problem. What was the exact cause, I am not sure. Thanks again for your help Nick
